### PR TITLE
Itempipes Returns

### PR DESCRIPTION
**Pull request by brachy84**
This pr introduces item pipes.

### How it works
When something tries to insert into a pipe, the net will gather all itemhandlers with distance to source and caches them for next transfer. The net then tries to insert the stack based on the distribution mode.
The distribution mode can be set on conveyors next to pipes in gui. The current modes are insert first and (true) round robin.
I have not tested it much, but it's working perfectly fine on the stuff i tested

There are 5 sizes.
Each size can transport different amount of items per sec.
The rate and also the max range can be set via material.
Pipes respect conveyor filters on insert
Pipes respect robot arm settings on insert
It does not use methods from robot arm. It mimics its behaviour
Robot arms can NOT insert into pipes. If it is something you really want, i can implement it.

ItemWalker is now only used to gather information about inventories of the net.
The transfer logic is now in ItemNetHandler

Also fixes pipe and cable hitbox

Cover gui when connected to ONE pipe
grafik
grafik
The first button toggles between round robin and first insert.
The second button toggles if the inventory should accept items if the cover is set to insert to pipes (to avoid looping)

### CODE REVIEW

**Review by Irgendwer01 — APPROVED** [review]
Looks fine to me

**Comment by brachy84** [comment]
Wait for the material meta item pr

**Comment by Irgendwer01** [comment]
> Wait for the material meta item pr
I need to wait tho sadly

File: src/main/resources/assets/gregtech/lang/en_us.lang
Comment (TechLord22) [inline src/main/resources/assets/gregtech/lang/en_us.lang:866]: I reccomend rephrasing this slightly to: "to pull items from the cover holder into pipes.

**Review by TechLord22 — CHANGES_REQUESTED** [review]

File: src/main/java/gregtech/api/unification/ore/OrePrefix.java
Comment (brachy84) [reply src/main/java/gregtech/api/unification/ore/OrePrefix.java:136]: gt5 just had for like 2 materials only 3 pipes. The rest was 5 pipes and 3 restrictive. gt6 has 5 normal and 5 restrictive, but i do agree that we should have less pipes. How about 4 normal (small - huge) and 2 restrictive (medium & large)

File: src/main/java/gregtech/common/pipelike/itempipe/net/ItemNetHandler.java
Comment (brachy84) [reply src/main/java/gregtech/common/pipelike/itempipe/net/ItemNetHandler.java:34]: um what? if it is ticking, cast it. if it is not ticking, set it to tick

File: src/main/java/gregtech/common/pipelike/itempipe/net/ItemNetWalker.java
Comment (brachy84) [reply src/main/java/gregtech/common/pipelike/itempipe/net/ItemNetWalker.java:27]: I will probably remove them. The walker walks pretty well

File: src/main/resources/assets/gregtech/lang/en_us.lang
Comment (brachy84) [reply src/main/resources/assets/gregtech/lang/en_us.lang:866]: will rename "cover holder" to "inventory" because the cover holder can be the pipe itself

**Review by TechLord22 — APPROVED** [review]

**Comment by brachy84** [comment]
I have found a major bug and I'm unable to find why it happens.
When the target inventory has a conveyor (or robot arm) on the opposite the pipe is attached to, then the items will not be inserted.
Only happens when the conveyor is set to export.
I will try to fix the next few days.

**Comment by brachy84** [comment]
I'm dead
I got the capality from the wrong side.

**Comment by brachy84** [comment]
Performance test:
grafik
I placed a 10x10 wall of bronze chests with pipes on both sides and lv cover (export) on one side.
I placed a aluminium chest full of dirt near the wall and connected it, so I have 2 pipe nets with 103 pipes each
(cover on aluminium chest is uv, export and round robin)
I did the same thing for eio conduits
While not working:
- eio: 190 - 195 FPS (lag spikes when placing and breaking conduits in a larger net)
- gt: 220 - 225 FPS

While working:
- eio: ca. 190 FPS
- gt: 200 - 220 FPS

GT item pipes have a slightly better performance. Now this is only 206 pipes. I would like to see a comparison with 1000 pipes

File: src/main/java/gregtech/api/pipenet/block/BlockPipe.java
Comment (DStrand1) [inline src/main/java/gregtech/api/pipenet/block/BlockPipe.java:388]:
```suggestion
        return canPipeConnectToBlock(selfTile, facing.getOpposite(), other);
```
Fixes an issue where sided capabilities will not properly connect. For example, the singleblock generators, or the dynamo hatch

**Comment by DStrand1** [comment]
As pointed out in Discord (but putting it here as well), fluid pipes seem to be basically non-functional with this PR. However, with the fix above, Cables seem to work fine, and in my (limited) testing, Item Pipes are also perfectly fine. I don't want Fluid Pipes to block this PR, so as long as you are willing to work on them after this is merged, it's fine by me. I could review the code a bit more, but I think it could be merged with the one above fix

**Comment by brachy84** [comment]
I did a bit of testing and cables are not fine
grafik
the first one fills almost up and the other two only have like 3000. They get a little bit after reconnecting pipes

**Pull request merged**
